app/models/policy_host.rb

```diff
@@ -8,10 +8,26 @@ class PolicyHost < ApplicationRecord
   validates :policy, presence: true
   validates :host, presence: true, on: :create
   validates :host_id, presence: true, uniqueness: { scope: :policy }
+  validate :host_supported?, on: :create
+
+  scope :supported_os_versions, lambda { |policy_id, host_os_major_version|
+    profile = Profile.find_by(policy_id: policy_id)
+    if host_os_major_version.to_s == profile&.os_major_version.to_s
```

Comment:
No, this should definitely not be part of the scope, ideally it should live in the `host_supported?` method.